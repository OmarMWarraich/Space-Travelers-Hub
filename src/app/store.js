import { configureStore } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
/* eslint-disable-next-line import/prefer-default-export */
export const store = configureStore({
  reducer: {
    counter: counterReducer,
  },
});
