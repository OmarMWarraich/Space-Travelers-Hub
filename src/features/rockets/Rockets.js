import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRocketsAsync } from './rocketsSlice';
import Rocket from './Rocket';

const Rockets = () => {
  const dispatch = useDispatch();

  const Rockets = useSelector((state) => state.rockets);
  console.log(Rockets);
  useEffect(() => {
    if (Rockets.length === 0) {
      dispatch(fetchRocketsAsync());
    }
    /* eslint-disable-next-line react-hooks/exhaustive-deps */
  }, []);

  return (
    <>
      <div>
        {Rockets.map((rocket) => (
          <Rocket key={rocket.id} rocket={rocket} />
        ))}
      </div>
    </>
  );
};

export default Rockets;
