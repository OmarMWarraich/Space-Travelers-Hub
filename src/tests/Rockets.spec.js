import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../app/store';
import Rockets from '../features/rockets/Rockets';

it('Rockets page rendering', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Rockets />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
