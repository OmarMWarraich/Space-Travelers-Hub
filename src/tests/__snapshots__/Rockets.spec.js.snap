// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets page rendering 1`] = `<div />`;
